### PR TITLE
[KTIJ-11416] Pasting multiline text to KDoc like in JavaDoc

### PR DESCRIPTION
**Pull request by i582**
# Pasting multiline text to KDoc like in JavaDoc

Code:

```kotlin
/**
 * <caret>
 */
fun main() {}
```

Paste text:

```text
Some
multiline
text
```

## Before

```kotlin
/**
 * Some
multiline
text
 */
fun main() {}
```

## After (like for JavaDoc)

```kotlin
/**
 * Some
 * multiline
 * text
 */
fun main() {}
```

Fixes [KTIJ-22293](https://youtrack.jetbrains.com/issue/KTIJ-22293) [marked as duplicate]
Fixes [KTIJ-11416](https://youtrack.jetbrains.com/issue/KTIJ-11416)

### CODE REVIEW

File: plugins/kotlin/plugin/k1/resources/META-INF/kotlin-core-fe10.xml [hunk 1]
```diff
@@ -193,6 +193,7 @@
     <copyPastePostProcessor implementation="org.jetbrains.kotlin.idea.conversion.copy.ConvertTextJavaCopyPasteProcessor"/>
     <copyPastePostProcessor implementation="org.jetbrains.kotlin.idea.codeInsight.KotlinCopyPasteReferenceProcessor"/>
     <copyPastePreProcessor implementation="org.jetbrains.kotlin.idea.editor.KotlinLiteralCopyPasteProcessor"/>
+    <copyPastePreProcessor implementation="org.jetbrains.kotlin.idea.editor.KDocCopyPastePreProcessor"/>
```
Comment (dimonchik0036) [inline plugins/kotlin/plugin/k1/resources/META-INF/kotlin-core-fe10.xml:196]: The code doesn't depend on resolve, so it should be located in `plugins/kotlin/plugin/common/resources/META-INF/kotlin-core.xml`

**Review by dimonchik0036 — CHANGES_REQUESTED** [review]
* The commit contains an obsolete link (to IDEA-298460 instead of KTIJ-11416)
* Missing tests for different cases

File: plugins/kotlin/plugin/k1/resources/META-INF/kotlin-core-fe10.xml (hunk 1 above)
Comment (i582) [reply plugins/kotlin/plugin/k1/resources/META-INF/kotlin-core-fe10.xml]: Fixed.

**Comment by i582** [comment]
@dimonchik0036 thanks for the review, added tests and fixed link in first commit

**Comment by i582** [comment]
@dimonchik0036 please note that since the last review I have redone the processing of pasting text with leading empty lines.

In Java it works like this:

Paste text:
```


Hello
World
```

Code:
```java
/**
 * <caret>
 */
```

Result:
```java
/**
 *

 Hello
 World
 */
```

Which seems a little strange to me. In the last commit, I changed this behavior for KDoc and now the result will be the same as for text with trailing empty lines, i.e. they will be cut off and only the text will be pasted:

```kotlin
/**
 * Hello
 * World
 */
```

**Review by dimonchik0036 — APPROVED** [review]
Thank you for the contribution! Will be merged manually

**Pull request closed without merging**